internal/common/backoff/retry.go

```diff
@@ -63,6 +65,16 @@ func (c *ConcurrentRetrier) throttleInternal(doneCh <-chan struct{}) time.Durati
 	c.Lock()
 	if c.failureCount > 0 {
 		next = c.retrier.NextBackOff()
+		// If secondary is included, use the greatest of the two (which also means
+		// if one is "done", which is -1, the one that's not done is chosen)
+		if c.includeSecondaryRetrier {
+			c.includeSecondaryRetrier = false
+			if c.secondaryRetrier != nil {
+				if secNext := c.secondaryRetrier.NextBackOff(); secNext > next {
```

Comment:
This ended up being exactly like the server code where the attempt isn't shared between the 2 retriers.
I think this is fine, it's just different from TS.